v03_pipeline/lib/tasks/base/base_variant_annotations_table.py

```diff
@@ -7,10 +9,30 @@
     variant_annotations_table_path,
 )
 from v03_pipeline.lib.tasks.base.base_update_task import BaseUpdateTask
-from v03_pipeline.lib.tasks.files import GCSorLocalTarget, HailTableTask
+from v03_pipeline.lib.tasks.files import GCSorLocalTarget
+from v03_pipeline.lib.tasks.reference_data.updated_reference_dataset_collection import (
+    UpdatedReferenceDatasetCollectionTask,
+)
 
 
 class BaseVariantAnnotationsTableTask(BaseUpdateTask):
+    @cached_property
```

Comment:
does `cached_property` make sense here? in my new task, we're reading the annotations table at least twice if the annotations table needs to be updated (in complete() and update_table())

Reply:
I think the idea is correct!  We can definitely leave it, however hail is lazy, so I generally haven't bothered caching `hl.read_table`... it's mostly just initializing the table in memory and not doing any actual meaningful work.

Reply:
makes sense, I just made it a property